### ISSUE TITLE
actionProductAdd doesn't pass id_product_old on duplicate product

### ISSUE THREAD

**Issue by alejoasotelo** [PrestaShop/PrestaShop#10678]
**Describe the bug**
The hook actionProductAdd doesn't pass the param id_product_old when duplicate product.
It's necessary to have the old id when duplicating a product in order to also be able to clone the module data.
I know I could do an override, but this isn't the best solution.

**To Reproduce**
1. Create a module with the hook 'actionProductAdd'.
2. Go to 'Product'
3. Select a product
4. Click on 'Duplicate'
5. The hook never pass the id of the old product.

**More info**
[(BOOM-4863) Duplicate product hook for modules - PrestaShop Issue Tracker](http://forge.prestashop.com/browse/BOOM-4863)

**Additionnal information**
PrestaShop version: 1.7.3.4
PHP version: 7.1

**Comment by khouloudbelguith** [comment]
Hi @alejoasotelo,

Thanks for your report.
The solution That I posted in this comment: 
http://forge.prestashop.com/browse/BOOM-4863?focusedCommentId=162626&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-162626 doesn't fix your issue?

Best regards, Khouloud

**Comment by alejoasotelo** [comment]
Hi @khouloudbelguith 
Thanks for your help!
Yes, I read it.
I should not modifiy the core or create an override to get the old id. That is a temporary solution. We need a permanent solution. 
The hook to duplicate product shold be include the id_product_old. What is the use of having the hook when a product is duplicated if the source product is not known?
Maybe 'actionProductAdd' should not include id_product_old, maybe PS needs a new hook called actionProductDuplicated o something like that.

Best!

**Comment by khouloudbelguith** [comment]
Hi @eternoendless,

To include id_product_old in the list of params of the "hook actionProductAdd". We need to add this parameter actionProductAdd in the file "SHOP_FOLDER/src/Adapter/Product/AdminProductDataUpdater.php" and replace L215 by this : 

```$this->hookDispatcher->dispatchWithParameters('actionProductAdd', array('id_product' => (int)$product->id,'id_product_old'=>$id_product_old, 'product' => $product));```

What do you think?
Thanks!

Best regards, Khouloud

**Comment by hamzanatek** [comment]
Hi @alejoasotelo, @khouloudbelguith 

you need to respect the best practice and not make changes directly in core files.
i suggest to use something like :

`public function hookActionProductAdd($params)
    {
        /**
         * @var $attribute Symfony\Component\HttpFoundation\ParameterBag
         */
        $attribute = $params['request']->attributes;
        $oldId = $attribute->get('id');
    }`

and after that you can  use  $oldId.

Thanks!

**Comment by jf-viguier** [comment]
It seems that this commit solve the issue : https://github.com/PrestaShop/PrestaShop/commit/6655cab883dcec1cae6f2dd58087fae1fc7e268b#diff-31a3a3e02b67c2e9064868bd3841ceab

But this fix is not included in 1.7.5.1. Is it planned for the next release ?

**Comment by PierreRambaud** [comment]
hey @hamzanatek, @khouloudbelguith was talking to @eternoendless which is the lead dev of this project, so if even the Core team can't touch core files to add a feature, what can we do? :sob:

**Comment by jf-viguier** [comment]
Yes indeed ! It should be great that every core team member mention it in their github profiles.

**Comment by PierreRambaud** [comment]
It's an open source project and for some we are also working on other projects.
And it's already defined :wink: 
![image](https://user-images.githubusercontent.com/1462701/54374302-1d6d3d00-467f-11e9-999e-a078f63fe3ca.png)
![image](https://user-images.githubusercontent.com/1462701/54374316-2bbb5900-467f-11e9-84b8-e3b34eff4062.png)

And @khouloudbelguith is in product team: 
![image](https://user-images.githubusercontent.com/1462701/54374382-4e4d7200-467f-11e9-9ac9-3d164c982acc.png)

**Comment by jf-viguier** [comment]
Thanks for the answer but I don't have the "Member of something" in any of your profiles. Maybe it's not a public information or I do I have to activate something in github ?

**Comment by hamzanatek** [comment]
@jf-viguier im just giving the code that can be used before including the commit 6655cab in the Prestashop core. and in my point of view there's no need to make a change in the code because the old product ID exists in the request :) 
and by the way im not a member of Prestashop core ;) I'm just trying to help

**Comment by akiyamaSM** [comment]
Thanks @hamzanatek!! I have applied your code and it works like a charm. And as you said you don't even need to reach to the core.

**Comment by alejoasotelo** [comment]
@hamzanatek the old id needs to be in the params because it is a hook for add/duplicated a product. It would be wrong if the hook that is executed when add or duplicate a product doesn't have the old id in the params.
i think so get it from the request it is just an alternative to solve the problem but the best way it is get it from the $params.
Best!

**Comment by PierreRambaud** [comment]
@matks Could you have a check on it, during the migration of the page, you can maybe add this?

**Comment by Hlavtox** [comment]
Not relevant anymore, old product page is gone.
